### ISSUE TITLE
Samples not always playing properly!

### ISSUE THREAD

**Issue by F483** [F483/tanzlove#6]
* [Source report](https://www.reddit.com/r/AndroidGaming/comments/7x25qp/made_my_first_gametoytool_a_euclidean_rhythm/du7c6bq/?st=jdoghcr9&sh=2103eb35)
 * [Additional info](https://www.reddit.com/message/messages/aw5w9s)
 * [Video of Bug](https://www.reddit.com/message/messages/aw5w9s)

**Comment by F483** [comment]
[also see](https://love2d.org/forums/viewtopic.php?t=84883)

**Comment by ghost** [comment]
The problem is that history can't go back, only forward, which isn't a problem when sys._getLoopLen() decreases; however, as sys._getLoopLen() increases, sys._setTotalProgress will set a bigger and bigger value, which takes longer and longer to catch up with.

This patch fixes it for me:

```diff
diff --git a/src/sys.lua b/src/sys.lua
index 193bf3b..ddb085b 100644
--- a/src/sys.lua
+++ b/src/sys.lua
@@ -181,6 +181,9 @@ end
 
 function sys._setTotalProgress(progress, deck)
     sys.player.clock[deck] = progress * sys._getLoopLen(deck)
+    for i = 1, sys.limits.tracks do
+        sys.player.history[deck][i] = -123456789.0
+    end
 end
 
 function sys.getRhythm(d, t)
```

**Comment by F483** [comment]
@pgimeno thank you very much! What you say makes total sense and this fixes the issue. Added your patch and will build an update asap.

FYI I will be rewriting the entire timing code to use a single mesure of beats played instead translating from time. Besides being the more elegant way of doing it, I will need it for when I make len track specific to allow for polyrhythms.

**Comment by F483** [comment]
Fix now rolled out in version 1.0.3
